iceoryx_posh/include/iceoryx_posh/popo/wait_set.hpp

```diff
@@ -26,129 +32,96 @@ class Condition;
-///
-/// @endcode
+/// The WaitSet stores Triggers and allows the user to wait till one or more of those Triggers are triggered. It works
+/// over process borders. With the creation of a WaitSet it requests a condition variable from RouDi and destroys it
+/// with the destructor. Hence the lifetime of the condition variable is bound to the lifetime of the WaitSet.
 class WaitSet
 {
   public:
-    using ConditionVector = cxx::vector<Condition*, MAX_NUMBER_OF_CONDITIONS_PER_WAITSET>;
+    using TriggerVector = cxx::vector<Trigger, MAX_NUMBER_OF_TRIGGERS_PER_WAITSET>;
+    using TriggerStateVector = cxx::vector<TriggerState, MAX_NUMBER_OF_TRIGGERS_PER_WAITSET>;
 
     WaitSet() noexcept;
     virtual ~WaitSet() noexcept;
+
+    /// @brief all the Trigger have a pointer pointing to this waitset for cleanup
+    ///        calls, therefore the WaitSet cannot be moved
     WaitSet(const WaitSet& rhs) = delete;
     WaitSet(WaitSet&& rhs) = delete;
     WaitSet& operator=(const WaitSet& rhs) = delete;
     WaitSet& operator=(WaitSet&& rhs) = delete;
 
-    /// @brief Adds a condition to the internal vector
-    /// @param[in] condition, condition to be attached
-    /// @return Returns an expected, that can contain on error from WaitSetError
-    cxx::expected<WaitSetError> attachCondition(Condition& condition) noexcept;
-
-    /// @brief Removes a condition from the internal vector
-    /// @param[in] condition, condition to be detached
-    /// @return True if successful, false if unsuccessful
-    void detachCondition(Condition& condition) noexcept;
-
-    /// @brief Clears all conditions from the waitset
-    void detachAllConditions() noexcept;
-
-    /// @brief Blocking wait with time limit till one or more of the condition become true
-    /// @param[in] timeout How long shall be waited for a signalling condition
-    /// @return ConditionVector vector of condition pointers that have become
-    /// fulfilled
-    ConditionVector timedWait(const units::Duration timeout) noexcept;
-
-    /// @brief Blocking wait till one or more of the condition become true
-    /// @return ConditionVector vector of condition pointers that have become
-    /// fulfilled
-    ConditionVector wait() noexcept;
-
-    /// @brief Is a given condition attached to this WaitSet
-    /// @return true if the condition is attached otherwise false
-    bool isConditionAttached(const Condition& condition) noexcept;
+    /// @brief Acquires a trigger from the waitset  The trigger is then attached to a class, the origin,
+    ///        which triggers the Trigger if a specific event happens. The class must then signal the
+    ///        trigger that it was triggered via the triggerCallback. If the WaitSet goes out of scope
+    ///        before the class does it calls the invalidationCallback to invalidate the Trigger inside
+    ///        of the class
+    ///        You cannot acquire an already logical equal acquired trigger. This means if you acquire a trigger
```

Comment:
You are right but I wanted to give the equivalence relation a name since you would assume intuitively that equivalence means that all members of both arguments (lhs, rhs) are the same which is not the case here.